pycloudlib/lxd/cloud.py

```diff
@@ -511,3 +494,220 @@ def _find_image(self, release, arch=LOCAL_UBUNTU_ARCH, daily=True):
+            profile_list: list, optional, profile(s) to use
+            user_data: used by cloud-init to run custom scripts/configuration
+            config_dict: dict, optional, configuration values to pass
+
+        Returns:
+            A list of string representing the command to be run to
+            launch the LXD instance.
+        """
+        if not profile_list:
+            profile_list = self._build_necessary_profiles(release=release)
+
+        cmd = super()._prepare_command(
+            name=name,
+            release=release,
+            ephemeral=ephemeral,
+            network=network,
+            storage=storage,
+            inst_type=inst_type,
+            profile_list=profile_list,
+            user_data=user_data,
+            config_dict=config_dict
+        )
+
+        cmd.append("--vm")
+
+        return cmd
+
+    def _get_image_hash_key(self, release=None):
+        """Get the correct hash key to be used to launch LXD instance.
+
+        When query simplestreams for image information, we receive a
+        dictionary of metadata. In that metadata we have the necessary
+        information to allows us to launch the required image. However,
+        we must know which key to use in the metadata dict to allows
+        to launch the image.
+
+        Args:
+            release: string, optional, Ubuntu release
+
+        Returns
+            A string specifying which key of the metadata dictionary
+            should be used to launch the image.
+        """
+        return self.VM_HASH_KEY
+
+    def _search_for_image(
+        self, remote, daily, release, arch=LOCAL_UBUNTU_ARCH
+    ):
+        """Find the LXD fingerprint in a given remote.
+
+        Args:
+            remote: string, remote to prepend to image_id
+            daily: boolean, search on daily remote
+            release: string, Ubuntu release to look for
+            arch: string, architecture to use
+
+        Returns:
+            string, LXD fingerprint of latest image
+
+        """
+        if release == "xenial":
+            # xenial needs to launch images:ubuntu/16.04/cloud
+            # because it contains the HWE kernel which has vhost-vsock support
+            return self.XENIAL_IMAGE_VSOCK_SUPPORT
+
+        if release == "trusty":
+            # trusty is not supported on LXD vms
+            raise UnsupportedReleaseException(
+                release="trusty",
+                is_vm=True
+            )
+
+        return super()._search_for_image(
+            remote=remote,
+            daily=daily,
+            release=release,
+            arch=arch
+        )
+
+    def image_serial(self, image_id):
+        """Find the image serial of a given LXD image.
+
+        Args:
+            image_id: string, LXD image fingerprint
+
+        Returns:
+            string, serial of latest image
+
+        """
+        if image_id == self.XENIAL_IMAGE_VSOCK_SUPPORT:
+            return ""
```

Comment:
Why `""` instead of `None`?

Reply:
To be honest, I thought that maybe one of ours scripts were always expecting a string to be returned here, but I am not sure if that is okay. @paride would it be okay if this method returns `None`  for xenial images on vms ?

Reply:
@lucasmoura the bootspeed measurement script does indeed expect a string, but I agree with Dan in that it should handle `None`, so +1 for Dan's suggestion. (In practice this is a non-issue as the bootspeed jobs do not measure Xenial, but I like to know it's working in case we want to enable it.)